context/app/api/client.py

```diff
@@ -165,45 +238,99 @@ def get_provenance(self, uuid):
-            "layers": [
-                {
-                    "name": "cells",
-                    "type": "CELLS",
-                    "url": cellsUri,
-                    "requestInit": {
-                        "headers": {
-                            'Authorization': 'Bearer ' + token
-                        }
-                    }
-                },
-            ],
-            "name": "Linnarsson",
-            "staticLayout": [
-                {
-                    "component": "scatterplot",
-                    "props": {
-                        "mapping": "UMAP",
-                        "view": {
-                            "zoom": 4,
-                            "target": [0, 0, 0]
+            return {
+                'description': 'DEMO',
+                'layers': [
+                    {
+                        'name': 'cells',
+                        'type': 'CELLS',
+                        'url': cellsUri,
+                        'requestInit': {
+                            'headers': {
+                                'Authorization': 'Bearer ' + token
+                            }
                         }
                     },
-                    "x": 0, "y": 0, "w": 12, "h": 2
-                },
-            ]
+                ],
+                'name': 'Linnarsson',
+                'staticLayout': [
+                    {
+                        'component': 'scatterplot',
+                        'props': {
+                            'mapping': 'UMAP',
+                            'view': {
+                                'zoom': 4,
+                                'target': [0, 0, 0]
+                            }
+                        },
+                        'x': 0, 'y': 0, 'w': 12, 'h': 2
+                    },
+                ]
+            }
+        else:
+            return self._build_vitessce_conf(entity)
+
+    def _build_assets_url(self, rel_path, uuid):
+        base_url = urllib.parse.urljoin(
+            current_app.config['ASSETS_ENDPOINT'],
+            str(Path(uuid) / Path(rel_path))
+        )
+        token_param = urllib.parse.urlencode({'token': self.nexus_token})
+        return base_url + '?' + token_param
+
+    def _build_image_schema(self, rel_path, uuid):
+        schema = {}
+        schema['name'] = Path(rel_path).name
+        schema['type'] = 'ome-tiff'
+        schema['url'] = self._build_assets_url(rel_path, uuid)
+        schema['metadata'] = {}
+        offsets_path = Path(CODEX_OFFSETS_PATH) / \
+            Path(Path(rel_path).name.replace('ome.tiff', 'offsets.json'))
+        schema['metadata']['omeTiffOffsetsUrl'] = self._build_assets_url(str(offsets_path), uuid)
+        return schema
+
+    def _build_image_layer_datauri(self, rel_path, uuid):
+        image_layer = {}
+        image_paths = [
+            str(rel_path).replace(
+                '#CODEX_TILE#', f'R{str(r).zfill(3)}_X{str(x).zfill(3)}_Y{str(y).zfill(3)}'
```

Comment:
Is this structure something that part of standard? Or just for the demo? Or somewhere in between?

Name could be more clear... perhaps `path_template` for `rel_path`?

Reply:
I think it's not going to always be a template.  It is more frequently going to actually be a relative path (CODEX is the only one that I know of - the scRNASEQ is not a template but the actual relative path.  This is what the ES response calls it).